components/layout_thread/lib.rs

```diff
@@ -515,6 +521,7 @@ impl LayoutThread {
                 error_reporter: self.error_reporter.clone(),
                 local_context_creation_data: Mutex::new(local_style_context_creation_data),
                 timer: self.timer.clone(),
+                quirks_mode: self.quirks_mode.unwrap_or(QuirksMode::NoQuirks),
```

Comment:
I think we do. I don't think we can layout anything without a document anyway. Indeed the sanest stuff would be making `build_shared_layout_context` taking the document directly, and don't store the quirks mode in the script thread at all.